web/src/settings_profile_fields.js

```diff
@@ -29,6 +29,111 @@ function display_success_status() {
+        $container.find("#alphabetize-choices-btn").prop("disabled", true).hide();
+    } else {
+        $container.find("#alphabetize-choices-btn").prop("disabled", false).show();
+    }
+}
+
+function sorting_field_choices_data(field_data) {
+    const sorted_data_array = Object.entries(field_data).sort((a, b) =>
+        a[1].text.toLowerCase().localeCompare(b[1].text.toLowerCase()),
+    );
+
+    const sorted_data = {};
+    let index = 1;
+    for (const [key, value] of sorted_data_array) {
+        value.order = index.toString();
+        sorted_data[key] = value;
+        index = index + 1;
+    }
+
+    return sorted_data;
+}
+
+function alphabetize_choices() {
+    const field_data = [];
+
+    // Collect input values into an array
+    $(".new-profile-field-form")
+        .find("div.choice-row input")
+        .each(function () {
+            const text = $(this).val().trim();
+            if (text) {
+                field_data.push(text);
+            }
+        });
+
+    // Sort the array
+    field_data.sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
+
+    // Update input values with sorted array
+    $(".new-profile-field-form")
+        .find("div.choice-row input")
+        .each(function (index) {
+            $(this).val(field_data[index] || ""); // Set empty string if no corresponding value in sorted array
+        });
+
+    $("#alphabetize-choices-btn").prop("disabled", true).hide();
+}
+
+function alphabetize_choices_edit_form($profile_field_form) {
+    let field_data = {};
+    field_data = read_select_field_data_from_form($profile_field_form);
+
+    const alphabetized_field_data = sorting_field_choices_data(field_data);
+
+    const $choice_list = $profile_field_form.find(".edit_profile_field_choices_container");
+    $choice_list.off();
+    $choice_list.empty();
+
+    const choices_data = parse_field_choices_from_field_data(alphabetized_field_data);
+
+    for (const choice of choices_data) {
+        $choice_list.append(
+            render_settings_profile_field_choice({
+                text: choice.text,
+                value: choice.value,
+            }),
+        );
+    }
+
+    // Add blank choice at last
+    create_choice_row($choice_list);
+    Sortable.create($choice_list[0], {
+        onUpdate: () => toggle_alphabetize_btn($(".profile-field-form")),
+        filter: "input",
+        preventOnFilter: false,
+    });
+
+    $("#alphabetize-choices-btn").prop("disabled", true).hide();
+    $profile_field_form
+        .find(".edit_profile_field_choices_container")
+        .on("input", ".choice-row input", (e) => {
+            add_choice_row(e);
+            toggle_alphabetize_btn($(".profile-field-form"));
+        });
+    $profile_field_form
+        .find(".edit_profile_field_choices_container")
+        .on("click", "button.delete-choice", (e) => {
+            delete_choice_row(e);
+            toggle_alphabetize_btn($(".profile-field-form"));
+        });
+}
```

Comment:
This is almost same as `set_up_select_field_edit_form`. Can you try to deduplicate the code?

Reply:
I have tried deduplicate the code, by dividing the common code into functions ```render_field_choices``` and ```set_up_select_field_edit_form_events``` -- then using them in ```alphabetize_choices_edit_form``` and ```set_up_select_field_edit_form```

Reply:
Maybe these changes of extracting these into functions can be done in a separate commit, so that this becomes somewhat easy to review.